src/PhpSlackBot/Bot.php

```diff
@@ -217,7 +217,7 @@ private function init(\Zend\Log\LoggerInterface $logger = null)
         }
         $this->wsUrl = $response['url'];
 
-        if (is_null($logger)) {
+        if (is_null($this->logger) && is_null($logger)) {
```

Comment:
Correct me if I am wrong but $logger here is always null.
The init() method is always called without argument.

Reply:
Yeah that is right that init is always called init(), but you can set a logger before init gets called like in my code below...

```
        $logger = new Logger();
        $writer = new Stream("php://output");

        $filter = new Priority(Logger::CRIT);
        $writer->addFilter($filter);
        $logger->addWriter($writer);

        $bot = new Bot();
        $bot->setToken(config('slail.token'));
        $bot->initLogger($logger);
        $bot->loadCommand(new TailCatchallCommand());
        $bot->run();
```
In this code, I overrider the logger calling `initLogger($logger)`, then when the $bot->init() runs, it needs to make sure that a logger has not been set yet, and if not it can do the default.

Does this make sense now?

Reply:
Yes but maybe it would be better to do the following then:

Remove the argument in the init method
```php
private function init(\Zend\Log\LoggerInterface $logger = null)
```
```php
private function init()
```

And simplify the condition in this init method :
```php
if (is_null($this->logger) && is_null($logger)) {
    $this->initLogger();
}
```
```php
if (is_null($this->logger)) {
    $this->initLogger();
}
```

What do you think?